pkg/daemon/daemon.go

```diff
@@ -155,9 +171,19 @@ func (d *daemon) getIbSriovNetwork(networkID string) (string, *utils.IbSriovCniS
 		return "", nil, fmt.Errorf("failed to parse network id %s with error: %v", networkID, err)
 	}
 
-	netAttInfo, err := d.kubeClient.GetNetworkAttachmentDefinition(networkNamespace, networkName)
-	if err != nil {
-		return "", nil, fmt.Errorf("failed to get networkName attachment %s with error: %v", networkName, err)
+	// Try to get net-attach-def in backoff loop
+	var netAttInfo *v1.NetworkAttachmentDefinition
+	if err = wait.ExponentialBackoff(backoffValues, func() (bool, error) {
+		netAttInfo, err = d.kubeClient.GetNetworkAttachmentDefinition(networkNamespace, networkName)
+		if err != nil {
+			log.Warn().Msgf("failed to get networkName attachment %s with error %v",
+				networkName, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return "", nil, errcode.Errorf(ErrGetNetAttachDef, "failed to get networkName attachment %s",
```

Comment:
should we also add the last error to the returned error for completeness ?

Reply:
We discussed it week(?) ago and decided that warning inside backoff loop is enough.

Reply:
ok